_posts/2022-06-18-zio-streams.md

```diff
@@ -0,0 +1,537 @@
+---
+title: "ZIO Streams: An Introduction"
+date: 2022-06-12
+header:
+  image: "/images/blog cover.jpg"
+tags: [zio, zio-streams]
+excerpt: "An Introduction to ZIO Streams"
+---
+
+In this post, we're going to go over an introduction to the main components of
+ZIO Streams, how to work with them when things go right, and what to do when
+things go wrong.
+
+Additionally, we're going to go over a toy example of working with ZIO Streams
+asynchronously to get us in the async mind-set, as well as provide a simple, but
+functioning example of processing files modeled after markdown blog posts.
+
+## Set up
+
+We're going to base this discussion off of the latest ZIO 2.0 code, which is
+still an `RC`, but hopefully won't change before release. These are the
+dependencies to include for our walk through:
+
+```scala
+libraryDependencies ++= Seq(
+      "dev.zio" %% "zio" % "2.0.0-RC6",
+      "dev.zio" %% "zio-streams" % "2.0.0-RC6",
+      "dev.zio" %% "zio-json" % "0.3.0-RC8"
+)
+```
+
+## ZIO Stream Components
+
+Before we start discussing the core components of ZIO Streams, let's first
```

Comment:
I had originally wrote a small thing about Scala's LazyList, and then though it might just be confusing to then shift gears to zio streams. I went back and added in what may be more of a "medium" introduction.